acceptance/openstack/blockstorage/v3/blockstorage.go

```diff
@@ -79,6 +79,42 @@ func CreateVolume(t *testing.T, client *gophercloud.ServiceClient) (*volumes.Vol
 	return volume, nil
 }
 
+// CreateVolumeWithType will create a volume of the given volume type
+// with a random name and size of 1GB. An error will be returned if
+// the volume was unable to be created.
+func CreateVolumeWithType(t *testing.T, client *gophercloud.ServiceClient, vt *volumetypes.VolumeType) (*volumes.Volume, error) {
```

Comment:
This could be incorporated into `CreateVolume` but i opted into creating a new func for this. Let me know if its better to incorporate it instead.

Reply:
A new function is a good idea. This keeps the testing functions specific to only one type of resource.